### ISSUE TITLE
FEAT: Way to update state of table structure

### ISSUE THREAD

**Issue by LanKing** [dbgate/dbgate#666]
If table structure was changed in the background, there is no way to update state in UI. Needed some 🔃 UI control.

**Comment by janproch** [comment]
You could use this refresh button:

![image](https://github.com/dbgate/dbgate/assets/2543891/1e49e1b0-076e-44db-aef6-8ff6a6016111)

**Comment by anothergituser** [comment]
I have the opposite problem :) I would like to disable auto refresh because if you scroll all the way down, when it refreshed it jumps back up

**Comment by shenxgan** [comment]
if add a new index of postgresql table,  the table structure is not change even if click refresh button

**Comment by nilsso** [comment]
> if add a new index of postgresql table, the table structure is not change even if click refresh button

Just noticed this myself. Furthermore after committing an alter table all of the indexes on the table disappear from the table structure.

**Comment by janproch** [comment]
OK, this seems like a bug with incremental refresh

Until this is fixed, you could use Full structure refresh (at first click on "Connected" on status bar)

![Image](https://github.com/user-attachments/assets/b6469990-587f-4478-b209-d48fe349b76e)

**Comment by nilsso** [comment]
Yep that works for now. Thanks for pointing out the sync status, had no idea about it.
